state/event_table_test.go

```diff
@@ -536,6 +537,38 @@ func TestChunkify(t *testing.T) {
 	}
 }
 
+func TestChunkify2(t *testing.T) {
+	// Make 100 dummy events
+	events := make([]Event, 100)
+	for i := 0; i < len(events); i++ {
+		events[i] = Event{
+			NID: int64(i),
+		}
+	}
+
+	for _, tc := range chunkifyTestCases {
+		testCase := tc
+		t.Run(testCase.name, func(t *testing.T) {
+			chunks := sqlutil.Chunkify2[Event](testCase.numParamsPerStmt, testCase.maxParamsPerCall, events)
+			if len(chunks) != len(testCase.chunkSizes) {
+				t.Fatalf("got %d chunks, want %d", len(chunks), len(testCase.chunkSizes))
+			}
+			eventNID := int64(0)
+			for i := 0; i < len(chunks); i++ {
+				if len(chunks[i]) != testCase.chunkSizes[i] {
+					t.Errorf("chunk %d got %d elements, want %d", i, len(chunks[i]), testCase.chunkSizes[i])
+				}
+				for j, ev := range chunks[i] {
+					if ev.NID != eventNID {
+						t.Errorf("chunk %d got wrong event in position %d: got NID %d want NID %d", i, j, ev.NID, eventNID)
```

Comment:
It would be nice to reuse the code in TestChunkify.

Reply:
In the medium term I'd probably move everything over to the generic version so that there's only one true Chunkify.